packages/common/src/utils/formatUtil.ts

```diff
@@ -34,6 +34,36 @@ export const formatCount = (count: number) => {
   }
 }
 
+/**
+ * The format any currency should be:
+ * - show 0 if 0
+ * - don't show decimal places if input is a round number
+ * - show only up to 2 decimal places if input is not a round number
+ * - round down to nearest thousand if input is greater than 10000
+ * ie.
+ * 0 => 0
+ * 8 => 8
+ * 8.01 => 8.01
+ * 8.10 => 8.10
+ * 4,210 => 4210
+ * 9,999.99 => 9999.99
+ * 56,010 => 56K
+ * 443,123 => 443K
+ */
+export const formatCurrencyBalance = (amount: number) => {
```

Comment:
I know our test framework is mildly broken at the moment, but this seems like a great thing to write test cases for if you're up for it.

Reply:
gonna follow up with this.